### PR TITLE
Create a simple "integration benchmark" test for Firestore

### PR DESCRIPTION
**Pull request by jskeet**
This is designed to help diagnose #3182.

### CODE REVIEW

File: apis/Google.Cloud.Firestore/Google.Cloud.Firestore.IntegrationBenchmarks/Program.cs [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+            var histogram = new LongConcurrentHistogram(1L, TimeStamp.Minutes(10), 4);
+
+            long documentCounter = 0;
+            var data = new byte[config.DataSize];
+            RandomNumberGenerator.Create().GetBytes(data);
+
+            // Warm the pool up with no throttling
+            SemaphoreSlim qpsThrottle = new SemaphoreSlim(config.PoolSize);
+            for (int i = 0; i < config.PoolSize; i++)
+  	        {
```
Comment (chrisdunelm) [inline apis/Google.Cloud.Firestore/Google.Cloud.Firestore.IntegrationBenchmarks/Program.cs]: nit: indenting

File: apis/Google.Cloud.Firestore/Google.Cloud.Firestore.IntegrationBenchmarks/Program.cs [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+
+            async Task WriteBatchesAsync()
+            {
+                while (Interlocked.Decrement(ref batchesLeft) >= 0)
+                {
+		            await WriteBatchAsync();
+                }
+            }
+	    
+	        async Task WriteBatchAsync()
```
Comment (chrisdunelm) [inline apis/Google.Cloud.Firestore/Google.Cloud.Firestore.IntegrationBenchmarks/Program.cs]: nit: indenting

File: apis/Google.Cloud.Firestore/Google.Cloud.Firestore.IntegrationBenchmarks/Program.cs [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+
+            Task.WaitAll(tasks);
+            Interlocked.Increment(ref completed);
+            return histogram;
+
+            async Task WriteBatchesAsync()
+            {
+                while (Interlocked.Decrement(ref batchesLeft) >= 0)
+                {
+		            await WriteBatchAsync();
```
Comment (chrisdunelm) [inline apis/Google.Cloud.Firestore/Google.Cloud.Firestore.IntegrationBenchmarks/Program.cs]: nit: indenting

**Comment by jskeet** [comment]
Thanks for catching the nits - those were due to editing on a VM, I suspect...

**Pull request merged**